src/development/platform-integration/ios/app-extensions.md

```diff
@@ -0,0 +1,213 @@
+a custom engine and embed the `FlutterViewController`
+as described in the following section.
+
+{{site.alert.note}}
+  This requires a custom build of the Flutter engine and
+  is only for advanced users.
+  To learn more, check out [Compiling the engine][].
+{{site.alert.end}}
+
+<!-- TO DO: replace link with a ghost PR -->
+1. Create a custom build of the Flutter engine that removes uses of
+   `sharedApplication` and corrects the path for the bundle.
+   Check out an [example from the community on GitHub][].
+
+1. Open the Flutter app project settings in Xcode to share build
+   configurations. 
+
+   In the **Info** tab,
+   under the **Configurations** expandable group,
+   expand the **Debug**, **Profile**, and **Release** entries.
+   For each, select the same value from the drop-down menu for the
+   extension target as the entry selected for the normal app target. 
+
+    <figure class="site-figure {{include.class}}">
+        <div class="site-figure-container">
+            <img src='/assets/images/docs/development/platform-integration/app-extensions/xcode-configurations.png'
+            height='300'>
+        </div>
+    </figure>
+
+1. If the generated extension code includes a storyboard file, delete it. 
+    In the **Info.plist** file delete the **NSExtensionMainStoryboard** property, 
+    and add add the **NSExtensionPrincipalClass** property with a value that matches the 
+    name of your `ViewController`. For example, in an iMessage extension you would use 
+    `MessageViewController`. 
+    
+    <figure class="site-figure {{include.class}}">
+        <div class="site-figure-container">
+            <img src='/assets/images/docs/development/platform-integration/app-extensions/extension-info.png'
+            height='300'>
+        </div>
+    </figure>
+
+
+1. Embed the `FlutterViewController` as described in
+   [Adding a Flutter Screen][]. For example, you can display a 
+   specific route in your Flutter app within an iMessage extension.
+
+    ```swift
+    //This attribute tells the compiler that this piece of Swift code can be accessed from Objective-C.
+    @objc(MessagesViewController)
+
+    class MessagesViewController: MSMessagesAppViewController {
+        override func viewDidLoad() {
+            super.viewDidLoad()
+            showFlutter()
+        }
+    
+        @objc func showFlutter() {
+            // Create a FlutterViewController with an implicit FlutterEngine
+            let flutterViewController = FlutterViewController(project: nil, initialRoute: "/ext", nibName: nil, bundle: nil)
+            present(flutterViewController, animated: true, completion: nil)
+        }
+    ```
+
+{{site.alert.important}}
+Commenting out `sharedApplication` disables
+many features in the Flutter framework.
+This hasn't been tested in App Store submissions
+and might not work for some app extensions.
+For example, home screen Widgets can't use some
+lower-level APIs needed to draw Flutter UI.
+{{site.alert.end}}
```

Comment:
**question:** This seems important, so why is it at the very end of the page? I see no other reference to `sharedApplication`, so is this something that an iOS developer should know? What about a novice developer? Are we missing something to connect the dots?

Reply:
It's only relevant for this specific section. Only advanced users, as mentioned in the note above